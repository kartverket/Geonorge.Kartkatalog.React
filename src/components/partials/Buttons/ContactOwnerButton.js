import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import style from './Buttons.scss'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export class ContactOwnerButton extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {

        // TODO styling
        if (this.props.metadata.ContactMetadata) {
            let email = this.props.metadata.ContactMetadata.Email
            let icon = <FontAwesomeIcon title="Kontakt dataeier" icon={['far', 'envelope']} key="icon" />;
            let buttonClass = style.btn;
            let textContent = React.createElement('span', { key: "textContent" }, 'Kontakt dataeier');

            let childElements = [icon, textContent];
            return React.createElement('a', { href: "mailto:" + email, className: buttonClass }, childElements);
        } else {
            let icon = <FontAwesomeIcon title="Kontakt dataeier" icon={['far', 'envelope']} key="icon" />
            let buttonClass = style.btn + ' disabled';
            let textContent = React.createElement('span', { key: "textContent" }, 'Kontakt dataeier');
            let childElements = [icon, textContent];
            return React.createElement('span', { className: buttonClass }, childElements);
        }
    }

}

ContactOwnerButton.propTypes = {
    metadata: PropTypes.object.isRequired
};

export default connect(null, null)(ContactOwnerButton);
