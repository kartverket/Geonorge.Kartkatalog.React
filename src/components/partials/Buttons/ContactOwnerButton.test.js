import React from 'react';
import { shallow } from "enzyme";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ContactOwnerButton } from './ContactOwnerButton';

function setupItemWithContactMetadata() {
    const metadata = {
        Uuid: '1234',
        ContactMetadata: {
            Email:'test'
        } 
    }
    const props = {
        metadata: metadata
    }
    const wrapper = shallow(<ContactOwnerButton {...props} />)

    return {
        props,
        wrapper
    }
}

function setupItemWithoutContactMetadata() {
    const metadata = {
        Uuid: '1234',
    }
    const props = {
        metadata: metadata
    }
    const wrapper = shallow(<ContactOwnerButton {...props} />)

    return {
        props,
        wrapper
    }
}


describe('ContactOwnerButton', () => {
    
    it('should render self', () => {
        const { wrapper } = setupItemWithContactMetadata()
        expect(wrapper).toMatchSnapshot();
    })

    it('Item with DistributionUrl', () => {
        const { wrapper } = setupItemWithContactMetadata()

        expect(wrapper.hasClass('btn btn-default')).toBe(true)
        expect(wrapper.prop("href")).toBe('mailto:test')
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')        
        expect(wrapper.find("span").first().html()).toContain('Kontakt dataeier')
    })

    it('Item without DistributionUrl', () => {
        const { wrapper } = setupItemWithoutContactMetadata()
        
        expect(wrapper.hasClass('btn btn-default disabled')).toBe(true)
        expect(wrapper.prop("href")).toBeUndefined()
        expect(wrapper.find(FontAwesomeIcon).first().prop("icon")).toContain('external-link-square')        
        expect(wrapper.find("span").first().html()).toContain('Kontakt dataeier')
    })
})
