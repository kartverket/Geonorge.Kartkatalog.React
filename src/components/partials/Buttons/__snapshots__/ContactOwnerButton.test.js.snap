// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactOwnerButton should render self 1`] = `ShallowWrapper {}`;
